service/fullnode/fullnode/fullnode.go

```diff
@@ -0,0 +1,208 @@
+package fullnode
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/zeromicro/go-zero/core/logx"
+	"gorm.io/gorm"
+
+	"github.com/bnb-chain/zkbnb-go-sdk/client"
+	"github.com/bnb-chain/zkbnb/core"
+	"github.com/bnb-chain/zkbnb/dao/block"
+	tx "github.com/bnb-chain/zkbnb/dao/tx"
+	"github.com/bnb-chain/zkbnb/types"
+)
+
+const (
+	DefaultL2EndPoint = "http://localhost:8888"
+	SyncInterval      = 100 * time.Millisecond
+)
+
+type Config struct {
+	core.ChainConfig
+	L2EndPoint      string
+	SyncBlockStatus int64
+	LogConf         logx.LogConf
+}
+
+type Fullnode struct {
+	config *Config
+	client client.ZkBNBClient
+	bc     *core.BlockChain
+
+	quitCh chan struct{}
+}
+
+func NewFullnode(config *Config) (*Fullnode, error) {
+	bc, err := core.NewBlockChain(&config.ChainConfig, "fullnode")
+	if err != nil {
+		return nil, fmt.Errorf("new blockchain error: %v", err)
+	}
+
+	l2EndPoint := config.L2EndPoint
+	if len(l2EndPoint) == 0 {
+		l2EndPoint = DefaultL2EndPoint
+	}
+
+	if config.SyncBlockStatus <= block.StatusProposing ||
+		config.SyncBlockStatus > block.StatusVerifiedAndExecuted {
+		config.SyncBlockStatus = block.StatusVerifiedAndExecuted
+	}
+
+	fullnode := &Fullnode{
+		config: config,
+		client: client.NewZkBNBClient(l2EndPoint),
+		bc:     bc,
+
+		quitCh: make(chan struct{}),
+	}
+	return fullnode, nil
+}
+
+func (c *Fullnode) Run() {
+	curHeight, err := c.bc.BlockModel.GetCurrentBlockHeight()
+	if err != nil {
+		panic(fmt.Sprintf("get current block height failed, error: %v", err.Error()))
+	}
+
+	curBlock, err := c.bc.BlockModel.GetBlockByHeight(curHeight)
+	if err != nil {
+		panic(fmt.Sprintf("get current block failed, height: %d, error: %v", curHeight, err.Error()))
+	}
+
+	ticker := time.NewTicker(SyncInterval)
+	defer ticker.Stop()
+
+	syncBlockStatus := c.config.SyncBlockStatus
+	for {
+		select {
+		case <-ticker.C:
+
```

Comment:
The ticker should work only when we catch up with the latest block of the remote, if the current block is far behind the remote, we should keep catching, not wait for the ticker.